src/domain/service/cart.service.ts

```diff
@@ -8,13 +9,30 @@ export class CartService {
     private readonly cartItemService: CartItemService
   ) {}
 
+  async calculateTotalPrice(cartId: number): Promise<number> {
+    // TODO: Use database to calculate total price or calculate it on the server
+
+    const cartItems = await this.cartItemService.findBy({ cartId });
+
+    let total = 0;
+
+    for (const item of cartItems) {
+      total += item.price * item.quantity;
+    }
+
```

Comment:
I think maybe you can optimize this operation with `cartItems.count() * item.price * item.quantity` instead of iterating through the `cartItems[]`

Reply:
Can u give an example for it in more details ?